packages/jaeger-ui/src/components/SearchTracePage/SearchForm.js

```diff
@@ -71,6 +73,93 @@ export function convTagsLogfmt(tags) {
   return JSON.stringify(data);
 }
 
+export function lookbackToTimestamp(lookback, from) {
+  const unit = lookback.substr(-1);
+  return (
+    moment(from)
+      .subtract(parseInt(lookback, 10), unit)
+      .valueOf() * 1000
+  );
+}
+
+export const lookbackOptions = [
+  {
+    label: 'Hour',
+    value: '1h',
+  },
+  {
+    label: '2 Hours',
+    value: '2h',
+  },
+  {
+    label: '3 Hours',
+    value: '3h',
+  },
+  {
+    label: '4 Hours',
+    value: '4h',
+  },
+  {
+    label: '6 Hours',
+    value: '6h',
+  },
+  {
+    label: '12 Hours',
+    value: '12h',
+  },
+  {
+    label: '24 Hours',
+    value: '24h',
+  },
+  {
+    label: '2 Days',
+    value: '2d',
+  },
+  {
+    label: '3 Days',
+    value: '3d',
+  },
+  {
+    label: '5 Days',
+    value: '5d',
+  },
+  {
+    label: '7 Days',
+    value: '7d',
+  },
+  {
+    label: '2 Weeks',
+    value: '2w',
+  },
+  {
+    label: '3 Weeks',
+    value: '3w',
+  },
+  {
+    label: '4 Weeks',
+    value: '4w',
+  },
+];
+
+export const optionsWithinMaxLookback = _memoize(
+  () => {
+    const maxLookback = getConfigValue('search.maxLookback');
+    const { value: maxLookbackValue } = maxLookback;
+    const now = new Date();
+    const minTimestamp = lookbackToTimestamp(maxLookbackValue, now);
+    const options = lookbackOptions.filter(({ value }) => lookbackToTimestamp(value, now) >= minTimestamp);
+    if (options[options.length - 1].value !== maxLookbackValue) {
```

Comment:
These needs to check the timestamp, not the value.
E.g.: If the user configures `'1w'` it shouldn't be added just because `'1w' !== '7d'`

Reply:
ended up slightly different, but handles cases of `'1w'` vs `'7d'`